quinn-proto/src/connection/pacing.rs

```diff
@@ -91,6 +108,48 @@ impl Pacer {
     }
 }
 
+/// Calculates a pacer capacity for a certain window and RTT
+///
+/// The basic idea is to emit a burst (of size `capacity`) in timer intervals
+/// which are reasonable from an application point of view.
+/// The QUIC stack runs in user-space and is not subject to any real-time
+/// guarantees. Therefore timer-precison is limited.
+///
+/// Too small burst intervals means we will never meet them since the timer
+/// inaccuracy is higher. If we miss the interval by more than 25%, we will
+/// lose that part of the congestion window.
```

Comment:
The thing here is that the pacer tokens can only refill up to the maximum capacity of the pacer. Since tokens regenerate 25% than the congestion window, one can miss the exact point when the capacity refilled by 25% and still use the full window. If you wait any longer you will still have the same amount of tokens available and can send the same amount of packets (burst size), but not more. So this part of the congestion window is "lost".

Reply:
Therefore the "not completely using the congestion window" part will mainly happen if the intervals are too short, and timers are not handled in time. If the intervals are too long we instead will simply make pacing less effective - burst sizes are bigger, and chances that something on the link will drop some of the packets will therefore be higher.

Reply:
I tried to update this now in a newer revision. Let me know if it helps.

Reply:
The bullet points really help IMO, thanks.